packages/react-button/src/components/Button/Button.types.ts

```diff
@@ -145,22 +151,29 @@ export type ButtonTokens = ColorTokens &
       larger?: string;
       largest?: string;
     };
-
-    pressed?: {
-      transform?: string;
-      transition?: string;
-    };
   };
 
 /**
  * {@docCategory Button}
  */
 export type ButtonVariants<TTokens = ButtonTokens> = {
-  root?: TTokens;
   block?: TTokens;
   circular?: TTokens;
-  iconOnly?: TTokens;
   ghost?: TTokens;
+  iconOnly?: TTokens;
+  loading?: TTokens;
   primary?: TTokens;
+  root?: TTokens;
+
+  size_large?: TTokens;
+  size_larger?: TTokens;
+  size_largest?: TTokens;
+  size_regular?: TTokens;
+  size_small?: TTokens;
+  size_smaller?: TTokens;
+  size_smallest?: TTokens;
+
   transparent?: TTokens;
+
+  [key: string]: TTokens | undefined;
```

Comment:
Should we really allow this? I think it undermines type safety.

Reply:
How would someone add their own variants?

Reply:
you're possibly right but we need a good story for how someone would extend.